### PR TITLE
Patch set for altering the Boot order 

### PR DESCRIPTION
**Pull request by PandaWei**
The patch set is associated with the boot sequence of VM.
The VFIO cases will refer to them.

### CODE REVIEW

**Comment by PandaWei** [comment]
Please refer to auotest/tp-libvirt#497 for case result.

File: virttest/utils_test/libvirt.py [hunk 1]
```diff
@@ -1334,6 +1334,33 @@ def create_hostdev_xml(pci_id, boot_order=0):
...
+    vmxml.remove_all_boots()
+    # prepare PCI-device XML with boot order
+    device_domain = pci_id.split(':')[0]
+    device_domain = "0x%s" % device_domain
+    device_bus = pci_id.split(':')[1]
+    device_bus = "0x%s" % device_bus
+    device_slot = pci_id.split(':')[-1].split('.')[0]
+    device_slot = "0x%s" % device_slot
+    device_function = pci_id.split('.')[-1]
+    device_function = "0x%s" % device_function
```
Comment (will-Do) [inline virttest/utils_test/libvirt.py]: A ```try...except``` block may needed

File: virttest/utils_test/libvirt.py [hunk 2]
```diff
@@ -1334,6 +1334,33 @@ def create_hostdev_xml(pci_id, boot_order=0):
     return hostdev_xml.xml
 
 
+def alter_boot_order(vm_name, pci_id, boot_order=0):
+    """
+    Alter the startup sequence of VM to PCI-device firstly
+
+    OS boot element and per-device boot elements are mutually exclusive,
+    It's necessary that remove all OS boots before setting PCI-device order
```
Comment (will-Do) [inline virttest/utils_test/libvirt.py]: Pls finish the docstring by adding params

Comment (PandaWei) [inline virttest/utils_test/libvirt.py]: @will-Do
Thanks for your comments.
update it.

File: virttest/utils_test/libvirt.py [hunk 3]
```diff
@@ -1334,6 +1334,40 @@ def create_hostdev_xml(pci_id, boot_order=0):
...
+        device_domain = pci_id.split(':')[0]
+        device_domain = "0x%s" % device_domain
+        device_bus = pci_id.split(':')[1]
+        device_bus = "0x%s" % device_bus
+        device_slot = pci_id.split(':')[-1].split('.')[0]
+        device_slot = "0x%s" % device_slot
+        device_function = pci_id.split('.')[-1]
+        device_function = "0x%s" % device_function
+    except IndexError:
+        raise error.TestFail("Fail to get PCI info, pci_id: %s" % pci_id)
```
Comment (Hao-Liu) [inline virttest/utils_test/libvirt.py]: Should we raise a TestError here? Because we expect this `pci_id` to be valid and this is not what we're testing.

Comment (PandaWei) [inline virttest/utils_test/libvirt.py]: yes, looks more reasonable.
accept it.
Thanks

**Comment by Hao-Liu** [comment]
Merging this. Thanks @PandaWei @will-Do.

**Pull request merged**